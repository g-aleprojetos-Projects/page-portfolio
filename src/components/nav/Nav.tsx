import React, {useState} from 'react';
import {Hamburger} from 'components/hamburger';
import {RightNav} from 'components/rightNav';
import * as S from './Nav.styles';
import {useResponsiveContext} from 'context/mobileWindow';

export const Nav = () => {
  const {mobile} = useResponsiveContext();
  const [menuLateral, setMenuLateral] = useState<boolean>(false);

  const handleMudarMenu = () => {
    setMenuLateral(!menuLateral);
  };

  return (
    <S.Container data-testid={'test_nav'} mobile={mobile}>
      <S.Logo data-testid={'test_logo'} mobile={mobile}>
        Logo
      </S.Logo>
      <Hamburger openMenu={menuLateral} mudarEstado={handleMudarMenu} />
      <RightNav open={menuLateral} />
    </S.Container>
  );
};
