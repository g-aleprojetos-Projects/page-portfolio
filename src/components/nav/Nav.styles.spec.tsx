import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './Nav.styles';

describe('Header.styles', () => {
  test('Container DEVE ser igual ao snapshot se mobile for igual a "true"', () => {
    const container = renderer.create(<S.Container mobile={true} />).toJSON();
    expect(container).toMatchSnapshot();
  });
  test('Container DEVE ser igual ao snapshot se mobile for igual a "false"', () => {
    const container = renderer.create(<S.Container mobile={false} />).toJSON();
    expect(container).toMatchSnapshot();
  });
  test('Logo DEVE ser igual ao snapshot se mobile for igual a "true"', () => {
    const container = renderer
      .create(<S.ContainerLogo mobile={true} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });
  test('Logo DEVE ser igual ao snapshot se mobile for igual a "false"', () => {
    const container = renderer
      .create(<S.ContainerLogo mobile={false} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });
});
