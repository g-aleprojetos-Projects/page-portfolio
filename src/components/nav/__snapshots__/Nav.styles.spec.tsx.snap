// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.styles Container DEVE ser igual ao snapshot se mobile for igual a "false" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

<nav
  className="c0"
/>
`;

exports[`Header.styles Container DEVE ser igual ao snapshot se mobile for igual a "true" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

<nav
  className="c0"
/>
`;

exports[`Header.styles Logo DEVE ser igual ao snapshot se mobile for igual a "false" 1`] = `
.c0 {
  position: static;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 5.563rem;
  color: #ffffff;
  padding: 0 0.5rem 0 5.563rem;
}

<div
  className="c0"
/>
`;

exports[`Header.styles Logo DEVE ser igual ao snapshot se mobile for igual a "true" 1`] = `
.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 5.563rem;
  color: #ffffff;
  padding: 0;
}

<div
  className="c0"
/>
`;
